### PR TITLE
[coqdoc] Allow arbitrary titles to be set

### PR DESCRIPTION
**Pull request by Lysxia**
**Kind:** feature

Currently there is a `--set-subtitle` option, which only allows to set a subtitle:

```
(** * My.Long.Module: subtitle *)
```

With this, the title can also be set:

```
(** * Short.Module: subtitle *)
```

or

```
(** * Short.Module *)
```

Comments are welcome to decide whether this PR is worth finishing.

- [ ] Added / updated test-suite

- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [ ] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: tools/coqdoc/cpretty.mll [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
   let strip_eol s =
     let eol = s.[String.length s - 1] = '\n' in
-    (eol, if eol then String.sub s 1 (String.length s - 1) else s)
+    (eol, if eol then String.sub s 0 (String.length s - 1) else s)
```
Comment (Lysxia) [inline tools/coqdoc/cpretty.mll]: This looks like a bug that I fixed.

Comment (herbelin) [reply tools/coqdoc/cpretty.mll:66]: Looks indeed like a bug.

**Review by herbelin — APPROVED** [review]
Looks interesting and convenient!

I did not know about the subtitle feature actually. This PR is thus a (backward-compatible) generalization of it and I find it good and carefully written.

I have otherwise an additional side design question: the use of a title/subtitle is here controlled by the command line, but if we have a bunch of files treated the same, with a Makefile rule, or several files on the command line, I feel that we may also want to indicate the title with an annotation directly in the file, maybe say using `(** title rest-of-the-line-used-as-title-possibly-including-subtitle *)` as first coqdoc comment. Don't know what you think about it. [Not that I'm not a big fan of comments starting with special coqdoc words like `printing` because it may not scale well but, in the `title` case, it seems to be rather natural.]

I'm assigning.

PS: Sorry for the very long delay in reacting to the PR.

**Comment by Lysxia** [comment]
Thanks for the review! I like your suggestion actually. I was trying to be cautious at the time but now I'm more familiar with the state of the project, rather than hiding behind a command-line flag, baking such a feature in seems much more reasonable. I'll think about it again and come back with more ideas.

**Pull request closed without merging**